### PR TITLE
Simplify cmake build by providing hints on how to install dependencies

### PR DESCRIPTION
**Pull request by nakulkorde**
Check whether all the necessary dependencies are installed, if not notify the user accordingly.
Also, display how to install the respective packages in DEB/RPM environments.

### CODE REVIEW

File: cmake/modules/dependency_notifier.cmake [hunk 1]
```diff
@@ -0,0 +1,74 @@
+if(NOT CMAKE_C_COMPILER)
```
Comment (r-rojo) [inline cmake/modules/dependency_notifier.cmake]: Missing copyright text

File: cmake/modules/dependency_notifier.cmake [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    RPM: yum install gcc-c++") 
+endif()
+
+if(NOT libjson-c_FOUND)
+    message("-- json-c not found. Please install json-c package for you respective distribution:
+    DEB: apt-get install libjson0-dev
+    RPM: yum install json-c
+   If you have already installed this package in a nonstandard location 
+   please specify the location by defining the variable LIBJSON-C_ROOT in 
+   your cmake command as follows: cmake .. -DLIBJSON-C_ROOT=<path to install location>")
```
Comment (r-rojo) [inline cmake/modules/dependency_notifier.cmake]: This assumes the user is running cmake from a sub-directory of the git repo root. Maybe change `..` to `<path to clone dir>`?

File: cmake/modules/dependency_notifier.cmake [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    DEB: apt-get install uuid-dev
+    RPM: yum install libuuid
+   If you have already installed this package in a nonstandard location 
+   please specify the location by defining the variable LIBJSON-C_ROOT in 
+   your cmake command as follows: cmake .. -DUUID_ROOT=<path to install location>")
+endif()
+
+find_package(PythonInterp 2.7 REQUIRED)
+if(NOT PYTHONINTERP_FOUND )
+    message("-- No suitable python interpreter found. Please install python > 2.7 to satisfy dependency for building tools.")
```
Comment (r-rojo) [inline cmake/modules/dependency_notifier.cmake]: Text should be:
`... install Python >= 2.7 ...`

File: cmake/modules/libraries_config.cmake [hunk 4]
```diff
@@ -1,81 +1,81 @@
...
+check_library_exists(rt clock_gettime "" HAVE_LIBRT)
+
+# uuid check
+find_package(UUID)
+
+# json-c check
+find_package(json-c REQUIRED)
+
+# ncurses check: CURSES_FOUND CURSES_INCLUDE_DIRS CURSES_LIBRARIES
+find_package(Curses)
```
Comment (r-rojo) [inline cmake/modules/libraries_config.cmake]: Can we remove find Curses?

File: cmake/modules/libraries_config.cmake [hunk 5]
```diff
@@ -1,81 +1,81 @@
...
+
+# rt check
+find_package(RT)
+check_library_exists(rt clock_gettime "" HAVE_LIBRT)
+
+# uuid check
+find_package(UUID)
+
+# json-c check
+find_package(json-c REQUIRED)
```
Comment (r-rojo) [inline cmake/modules/libraries_config.cmake]: Why is json-c REQUIRED and not UUID?

File: libopae/CMakeLists.txt [hunk 6]
```diff
@@ -26,6 +26,7 @@
 
 project(libopae-c)
 
+
```
Comment (r-rojo) [inline libopae/CMakeLists.txt]: Do we need this extra line? Can we remove it if not? This way this file will not be included in the list of files that have been changed in this PR.

**Comment by nakulkorde** [comment]
I will remove compiler_config.cmake but libraries_config.cmake is changed. for eg: removed the curses dependency.

**Pull request closed without merging**